src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -1183,25 +1195,36 @@ private void ExtendClientArea()
             }
             GetWindowRect(_hwnd, out var rcWindow);
 
-            if (_isClientAreaExtended && WindowState != WindowState.FullScreen)
+            if (_isClientAreaExtended && WindowState != WindowState.FullScreen && GetStyle().HasAllFlags(WindowStyles.WS_BORDER))
             {
                 var margins = UpdateExtendMargins();
                 DwmExtendFrameIntoClientArea(_hwnd, ref margins);
 
-                unsafe
+                // On Windows 11 21H2 and later, corners are configurable.
+                // When doing so, we need to make sure that DWM draws the non-client frame for that to work correctly.
+                if (OperatingSystem.IsWindowsVersionAtLeast(10, 0, 22000)) {
+                    SetWindowCornerPreference(DwmWindowCornerPreference.DWMWCP_ROUND);
```

Comment:
I think some users used SetWindowCornerPreference to redefine corners on windows.
This PR will break it, but I don't see a better solution.
Aside from adding new Win32Properties and using it, but that's for another PR.

Reply:
As for the PR, we have https://github.com/AvaloniaUI/Avalonia/pull/17354 already :)